test/cpp/api/functional.cpp

```diff
@@ -56,10 +56,26 @@ TEST_F(FunctionalTest, AvgPool2d) {
 }
 
 TEST_F(FunctionalTest, AvgPool3d) {
-  auto x = torch::ones({2, 5, 5, 5}, torch::requires_grad());
+  auto x = torch::ones({2, 5, 5, 5});
   auto y = F::avg_pool3d(x, AvgPool3dOptions(3).stride(2));
 
   ASSERT_EQ(y.ndimension(), 4);
   ASSERT_TRUE(torch::allclose(y, torch::ones({2, 2, 2, 2})));
   ASSERT_EQ(y.sizes(), torch::IntArrayRef({2, 2, 2, 2}));
 }
+
+TEST_F(FunctionalTest, CosineSimilarity) {
+  auto input1 = torch::tensor({{1, 2, 3}, {4, 5, 6}});
```

Comment:
Looks like we'll need to specify the type as `torch::kFloat` for both `input1` and `input2`. Sorry for not picking this out earlier.

Reply:
Thanks a lot for the catch! Just fixed it.